Documentation/eliminating-pre-builts.md

```diff
@@ -91,7 +233,85 @@ Entry in project.assets.json:
+
+```xml
+<ItemGroup>
+  <PackageReference Include="Microsoft.Build" Version="17.3.2" /> <!-- depends on 'System.Text.Json' version 6.0.0 -->
+  <PackageReference Include="Microsoft.CodeAnalysis.Features" Version="4.10.0-1.24067.21" /> <!-- depends on 'System.Text.Json' version 9.0.0-preview.1.24073.8 -->
+</ItemGroup>
+```
+
+This is a version mismatch. In this case, `System.Text.Json.6.0.0` and `System.Text.Json.9.0.0-preview.1.24073.8` are also transitive dependencies.
+
+#### Use Case: Version Conflict
+
+Your build has output the following error message:
+
+```text
+2 new packages used not in baseline! See report at ./artifacts/source-build/self/prebuilt-report/baseline-comparison.xml for more information. Package IDs are:
+  System.Text.Json.6.0.0
+  System.Text.Json.9.0.0-preview.1.24073.8
+```
+
+You look at `./eng/SourceBuildPrebuiltBaseline.xml` and determine that `System.Text.Json.*` is not listed as an exclusion. You then examine `./prebuilt-usage.xml` and see the following:
+
+```xml
+<UsageData>
+  <Usages>
+    <Usage Id="System.Text.Json" Version="6.0.0" File=".../SomeProject/project.assets.json">
+    <Usage Id="System.Text.Json" Version="9.0.0-preview.1.24073.8" File=".../SomeProject/project.assets.json">
+  </Usages>
+</UsageData>
+```
+
+Note that the `IsDirectDependency="true"` attribute is not present for these pre-builts.
+
+The `project.assets.json` file shows the following:
+
+```json
+{ 
+  "targets": {
+    "net9.0": {
+      "Microsoft.Build/17.3.2": {
+        "type": "package",
+        "dependencies": {
+          "System.Text.Json": "6.0.0"
+        }
+      },
+      "Microsoft.CodeAnalysis.Features/4.10.0-1.24067.21": {
+        "type": "package",
+        "dependencies": {
+          "System.Text.Json": "9.0.0-preview.1.24073.8"
+        },
+      }
+    }
+  }
+}
+```
+
+At this point, determine how to handle the dependencies by following At this point, determine how to handle the dependency by following [adding dependencies documentation](https://github.com/dotnet/source-build/blob/main/Documentation/sourcebuild-in-repos/new-dependencies.md#adding-dependencies). Based on what you determine, you will likely have to do one or more of the following:
+
+- **Update version -** Use a different versions of Microsoft.Build and/or Microsoft.CodeAnalysis.Features.
+- **Use Version.Details and Version.Props -**
+  - Investigate the repos for Microsoft.Build and Microsoft.CodeAnalysis.Features. They may need to flow in the live versions of System.Text.Json. In this case, add System.Text.Json as a package reference & update `./eng/Version.Details.xml` and `./eng/Versions.Props` for System.Text.Json in these repos.
```

Comment:
Many repositories in the source-build graph already use NuGet's Central Package Management feature. It would be good to mention that CPM can help with lifting and unifying transitive dependencies via the transitive pinning feature: https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management#transitive-pinning